src/helper/docs/mod.rs

```diff
@@ -2,31 +2,161 @@
 pub mod man;
 
 /// Cheat sheet helper.
-pub mod cheat;
+pub mod cheat_sh;
+/// eg page helper.
+pub mod eg;
 
 use crate::config::Config;
-use crate::error::Result;
-use crate::helper::docs::cheat::show_cheat_sheet;
+use crate::error::{Error, Result};
+use crate::helper::docs::cheat_sh::CheatDotSh;
+use crate::helper::docs::eg::Eg;
 use crate::helper::docs::man::show_man_page;
 use console::{style, Style, Term};
 use dialoguer::theme::ColorfulTheme;
 use dialoguer::Select;
 use std::io::Write;
+use std::process::{Command, Stdio};
+use ureq::Request;
+
+/// The `HelpProvider` trait defines essential methods for fetching help content related to commands from a provider.
+///
+/// Each provider that implements this trait should provide a default URL used to retrieve the command help content.
+/// However, it also allows specifying a custom URL to override the default one.
+///
+/// This trait is generic and not tied to any specific command help system such as man pages or POSIX documentation,
+/// instead it relies on the implementation to define how to fetch and mark up the content.
+///
+/// # Methods
+/// - `url`: Returns the default URL of the provider.
+/// - `build_req`: Uses the given command and URL to create an HTTP GET request.
+/// - `err_handle`: Handles possible request errors, such as a non-existent command page on the provider.
```

Comment:
I recommend renaming the methods as follows for more clarity:

- `build_req` -> `build_request`
- `err_handle` -> `handle_error`